terraform/infrastructure/azure/main.tf

```diff
@@ -214,10 +220,13 @@ resource "azurerm_network_security_group" "security_group" {
   tags                = local.tags
 
   dynamic "security_rule" {
-    for_each = concat(
-      local.ports,
-      [{ name = "nodeports", port = local.ports_node_range, priority = 200 }]
-    )
+    # we keep this rule for one last release since the azurerm provider does not 
+    # support moving security rules that are inlined (like this) to the external resource one.
+    # Even worse, just defining the azurerm_network_security_group without the 
+    # "security_rule" block will NOT remove all the rules but do nothing.
+    # TODO(@3u13r): remove the "security_rule" block in the next release after this code has landed.
+    # So either after 2.18 or after 2.17.X if cerry-picked release.
+    for_each = [{ name = "konnectivity", priority = 1000, port = 8132 }]
```

Comment:
Do I understand correctly that we don't need this rule anymore, but we need to migrate it before we can remove it?

Reply:
yes, we need to keep the block to signal to Azure that we want to move the security rules out of the dynamic block. When we remove this block, existing Constellations will keep the konnectivity security rule as long as they don't delete it manually. But since the block doesn't exist anymore, new Constellation will only have the rules we want.